themes/ddbasic/README.md

```diff
@@ -1,18 +1,210 @@
+
+A number of [practical breakpoints](https://github.com/ding2/ding2/tree/master/themes/ddbasic/sass/configuration/_grid-settings.scss) divides the styling into different device cases based on only one dimension, the screen width. Frontend development in contributed modules should use the established breakpoints, and tests should cover all breakpoint cases.
+
+Specific Drupal / DDBasic rule of thumbs:
+
+* Drupal structures are replicated as nested SCSS objects. 
+* Viewmodes are styled inside the parent entity object
+* Responsive specifications are styled inside the selector they override using the  ```@include media(..)``` mixin.
+* Specifications of viewmode rendering in views are placed inside entities using the ```&```-parent selector keeping styling of entities in same file.
+
+### Example - Nesting / Styling of a general entity
+
+In general nesting levels should be kept minimal, but certain patterns of nesting supports the Drupal object hierachy well. This example shows how entities are styled in general:
+
+```less
+
+// File: ddbasic/sass/components/<entity>/<bundle>.scss
+
+// Bundle specific styling
+.entity-bundle {
+
+    // Field styling
+    .field-name-field-xyz {}
+	
+    // View mode specific styling
+    &.view-mode-1 {
+
+        .field-name-field-xyz {}
+    
+        // Responsive specifications
+        @include media($tablet) {}
+
+        @include media($mobile) {}	
+
+    }
+    
+    &.view-mode-2 {}
+}
+```
+
+
+See the [news.scss](https://github.com/ding2/ding2/tree/master/themes/ddbasic/sass/components/node/news.scss) for a complete example styling of an entity.
+
+## Theme development
+
+[Gulp](http://gulpjs.com/) will assist in workflow automation for the theme.
+
+The following exampels takes place in the theme folder:
+
+```
+~$ cd DRUPAL/profiles/ding2/themes/ddbasic
+```
+ 
+
+Install [Node.js](https://github.com/joyent/node/wiki/Installing-Node.js-via-package-manager) if it is not already available on your platform.
+
+Install Gulp and other packages:
+
+```
+ ~$ npm install
+```
+
+### Watch source files
+
+```
+~$ gulp watch
+```
+
+### Compile the source files once
+
+```
+~$ gulp sass
+```
+
+### Compile KSS from SCSS
+
+KSS target folder is [sites/all/themes/ddbasic/styleguide (temp link)](http://msd.ding2.server003.b14cms.dk/profiles/ding2/themes/ddbasic/styleguide)
+
+```
+# Compile all sass into one css file
+~$ gulp sassprod
+# Compile KSS into the styleguide folder 
+~$ gulp kss
+```
+
+### SCSS lint
+
+The [lint settings](https://github.com/ding2/ding2/tree/master/themes/ddbasic/package.json
+) extends the [stylelint-config-standard](https://github.com/stylelint/stylelint-config-standard/blob/master/index.js) with a few exceptions towards flexibility.
+
+```
+~$ gulp validate-sass 
+```
+Pull requests are required to pass the linter.
```

Comment:
As far as I can tell the current version of the DDBasic theme does not pass this linter.

Reply:
If current stying fails linting, let's push this into a seperate issue.
I'm unsure about how the settings in the gulp file are pushed into CI, if that is what you mean.